l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -134,124 +129,150 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
-    function setEraPostLegacyBridgeUpgradeFirstBatch(uint256 _eraPostLegacyBridgeUpgradeFirstBatch) external onlyOwner {
-        require(eraPostLegacyBridgeUpgradeFirstBatch == 0, "ShB: eFPUB already set");
-        eraPostLegacyBridgeUpgradeFirstBatch = _eraPostLegacyBridgeUpgradeFirstBatch;
-    }
-
-    /// @dev This sets the first post upgrade batch for era, used to check old withdrawals
-    /// @param  _eraLegacyBridgeLastDepositBatch The the zkSync Era batch number that processes the last deposit tx initiated by the legacy bridge
-    /// @param _eraLegacyBridgeLastDepositTxNumber The tx number in the _eraLegacyBridgeLastDepositBatch of the last deposit tx initiated by the legacy bridge
-    function setEraLegacyBridgeLastDepositTime(
+    function initialize(
+        address _owner,
+        uint256 _eraPostDiamondUpgradeFirstBatch,
+        uint256 _eraPostLegacyBridgeUpgradeFirstBatch,
         uint256 _eraLegacyBridgeLastDepositBatch,
         uint256 _eraLegacyBridgeLastDepositTxNumber
-    ) external onlyOwner {
-        require(eraLegacyBridgeLastDepositBatch == 0, "ShB: eLOBDB already set");
-        require(eraLegacyBridgeLastDepositTxNumber == 0, "ShB: eLOBDTN already set");
-        eraLegacyBridgeLastDepositBatch = _eraLegacyBridgeLastDepositBatch;
-        eraLegacyBridgeLastDepositTxNumber = _eraLegacyBridgeLastDepositTxNumber;
-    }
-
-    /// @dev transfer tokens from legacy erc20 bridge or mailbox and set chainBalance as part of migration process
-    function transferFundsFromLegacy(address _token, address _target, uint256 _targetChainId) external onlySelf {
-        if (_token == ETH_TOKEN_ADDRESS) {
-            uint256 balanceBefore = address(this).balance;
-            IMailbox(_target).transferEthToSharedBridge();
-            uint256 balanceAfter = address(this).balance;
-            require(balanceAfter > balanceBefore, "ShB: 0 eth transferred");
-            chainBalance[_targetChainId][ETH_TOKEN_ADDRESS] =
-                chainBalance[_targetChainId][ETH_TOKEN_ADDRESS] +
-                balanceAfter -
-                balanceBefore;
-        } else {
-            uint256 balanceBefore = IERC20(_token).balanceOf(address(this));
-            uint256 legacyBridgeBalance = IERC20(_token).balanceOf(address(legacyBridge));
-            require(legacyBridgeBalance > 0, "ShB: 0 amount to transfer");
-            IL1ERC20Bridge(_target).transferTokenToSharedBridge(_token);
-            uint256 balanceAfter = IERC20(_token).balanceOf(address(this));
-            require(balanceAfter - balanceBefore >= legacyBridgeBalance, "ShB: wrong amount transferred");
-            chainBalance[_targetChainId][_token] = chainBalance[_targetChainId][_token] + legacyBridgeBalance;
+    ) external reentrancyGuardInitializer initializer {
+        require(_owner != address(0), "ShB owner 0");
+        _transferOwnership(_owner);
+        if (eraPostDiamondUpgradeFirstBatch == 0) {
```

Comment:
I am not sure when this logic should be processed. The `initializer` function would be called during the creation of the proxy and so `eraPostDiamondUpgradeFirstBatch` would be 0 there always

Reply:
Not on mainnet. We do not want to reset the values on mainnet, but when testing we want to set some values otherwise the values do deposits do not go through